Services/Dashboard/GlobalScreen/classes/PDLayoutProvider.php

```diff
@@ -0,0 +1,46 @@
+<?php
+use ILIAS\GlobalScreen\Scope\Layout\Provider\AbstractModificationProvider;
+use ILIAS\GlobalScreen\Scope\Layout\Provider\ModificationProvider;
+use ILIAS\GlobalScreen\ScreenContext\Stack\ContextCollection;
+use ILIAS\GlobalScreen\ScreenContext\Stack\CalledContexts;
+use ILIAS\GlobalScreen\Scope\Layout\Factory\MainBarModification;
+use ILIAS\UI\Component\MainControls\MainBar;
+
+/**
+ * Class PDLayoutProvider
+ *
+ * @author Nils Haagen <nils.haagen@concepts-and-training.de>
+ */
+class PDLayoutProvider extends AbstractModificationProvider implements ModificationProvider
+{
+    /**
+     * @var Collection | null
+     */
+    protected $data_collection;
+
+    /**
+     * @inheritDoc
+     */
+    public function isInterestedInContexts() : ContextCollection
+    {
+        return $this->context_collection->desktop();
+
+    }
+
+    public function getMainBarModification(CalledContexts $screen_context_stack) : ?MainBarModification
+    {
+
+        $this->data_collection = $screen_context_stack->current()->getAdditionalData();
+        if(!$this->data_collection->is(\ilDashboardGUI::DISENGAGE_MAINBAR, true)) {
+            return null;
+        }
+
+        return $this->globalScreen()->layout()->factory()->mainbar()
+            ->withModification(
+                function(MainBar $mainbar) : ?MainBar {
+                    return $mainbar->withActive($mainbar::NONE_ACTIVE);
```

Comment:
This will deactivate all tools, too, e.g. the online help right. I think this is not desired.

Reply:
Yes, this will disengage all slates, entries and tools. As far as I understood, this is desired behavior. 
However, I could probably make an explicit exception for help, and not close this, if engaged.
@yvseiler ?

Reply:
I see... Yes, indeed, the help is an exception here, as it must persist across pages because it can contain instructions that you still want to see even if you click in the dashboard.

Reply:
Will do. But keeping the help visible is a general issue and will be part of mainbar.js, then.
I'll implement in a different branch, if that is OK for you, @alex40724 ?